### PR TITLE
#2 constructor2 d dimensions explicit

### LINKED ISSUE #2: Matrix(double[][] A, int m, int n)
Construct a matrix quickly without checking arguments.

### PR DESCRIPTION
**Pull request by KaraBeason**
Closes #2.  Add constructor that creates m by n matrix from a 2D array, taking the array, m and n as arguments. Tests pass, check style passes.

### CODE REVIEW

File: Javatrix.java [hunk 1]
```diff
@@ -25,6 +25,50 @@
...
+    {
+        this.m = x;
+        this.n = y;
+        matrix = new double[m][n];
+        for (int i = 0; i < arr.length; i++)
+        {
+            for (int j = 0; j < arr[0].length; j++)
+            {
+                // If index not out of range for arr.
+                if (i < m && j < n)
```
Comment (haietza) [inline Javatrix.java:63]: The API suggests that arguments don't need to be checked. Do you think the loop should just test i < m and j < n and only copy those specified rows/cols from the provided array, versus attempting to copy the whole thing and having to test the arguments?

**Review by haietza — CHANGES_REQUESTED** [review]

File: Javatrix.java (hunk 1 above)
Comment (KaraBeason) [reply Javatrix.java:63]: In that case, is it the expected behavior that it throws an ArrayIndexOutOfBoundsException when given x and y greater than the dimensions of the array?  If so should I re-write my test for that case to expect an exception, or should I remove it?

Comment (haietza) [inline Javatrix.java]: I think what you have looks right - we check the row length during each iteration and if it doesn't equal the initial n, then we know they are all not the same length and can throw the exception.

Comment (haietza) [reply Javatrix.java:63]: Well, the API does not mention any exceptions, so I think the most efficient way to do it is how you have it implemented. Sorry - I was thinking about the case when m and n were less than the array's rows and columns, not when they were greater. I think the Javatrix class is good - let me review the tests below.

File: JavatrixTest.java [hunk 2]
```diff
@@ -24,7 +24,88 @@
...
+        n = 2;
+
+        // Array that will be truncated by the constructor.
+        matrix = new double[5][4]; 
+        // The array for comparison to constructor array.
+        double[][] expected = new double[m][n];
+
+        for (int i = 0; i < 5; i++)
+        {
+            for (int j = 4; j < n; j++)
```
Comment (haietza) [inline JavatrixTest.java]: Why is j starting at 4 here?

Comment (KaraBeason) [reply JavatrixTest.java]: That is most definitely a typo.  Thanks for catching :)

File: Javatrix.java (hunk 1 above)
Comment (KaraBeason) [reply Javatrix.java:63]: Ah, gotcha!  I see you what you mean.  Ok I will leave as is for now!   Thanks!

**Comment by KaraBeason** [comment]
@haietza is this good to be merged, or are changes still needed?  Thank you!

**Review by haietza — APPROVED** [review]
Sorry for the delay - looks good! Thanks!

**Comment by haietza** [comment]
@KaraBeason Do you think we should change this to not check the arguments, as Dr Fenwick discussed in class today?

**Comment by KaraBeason** [comment]
@haietza yes, I will push a fix up

**Pull request merged**